platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/NonAncientEventWindow.java

```diff
@@ -35,6 +36,12 @@ public class NonAncientEventWindow {
     public static NonAncientEventWindow INITIAL_EVENT_WINDOW = new NonAncientEventWindow(
             ConsensusConstants.ROUND_FIRST, ConsensusConstants.ROUND_FIRST, EventConstants.FIRST_GENERATION);
 
+    /**
+     * This static variable is temporary until we throw the switch permanently from minGenNonAncient to
+     * minRoundNonAncient
+     */
+    private static AtomicBoolean useBirthRoundForAncient = new AtomicBoolean(false);
```

Comment:
Instead of using static state, would it be possible to pass in another parameter to the constructor of this class? My concern is that using the this static object might make unit tests easy to mess up, e.g. tests that only pass when run in a specific order.

Reply:
just hit that, unit tests are messing up because of the static state.  I will pass in a constructor argument.  I think that will still work out since the initial event window should alway say every event is non-ancient whether compared by round or by min generation.